metadata-transformer/src/main/java/no/unit/nva/metadata/extractors/ContributorExtractor.java

```diff
@@ -0,0 +1,70 @@
+package no.unit.nva.metadata.extractors;
+
+import no.unit.nva.model.Contributor;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.NameType;
+import no.unit.nva.model.Role;
+import no.unit.nva.model.exceptions.MalformedContributorException;
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.Objects.isNull;
+
+public final class ContributorExtractor {
+    private static final Logger logger = LoggerFactory.getLogger(ContributorExtractor.class);
+
+    public static final Function<ExtractionPair, EntityDescription> apply = extractOrConsumeError();
+
+    @JacocoGenerated
+    private ContributorExtractor() {
+
+    }
+
+    private static Function<ExtractionPair, EntityDescription> extractOrConsumeError() {
```

Comment:
While I don't object to this code, it took longer to read than a method with one or two if statement would have.